broker_app.py

```diff
@@ -34,166 +35,164 @@
 
-# http://0.0.0.0:5000/schemas?high_level_entity=type&domain_entity=biomaterial&concrete_entity=donor_organism&latest&json
-# http://0.0.0.0:5000/schemas?url=${schemaUrl}&json&deref
-@app.route('/schemas', methods=['GET'])
-def get_schemas():
-    args = request.args
+            if not json_schema:
+                return response_json(HTTPStatus.OK, latest_schema)
 
-    # params
-    url = args.get('url')
-    high_level_entity = args.get('high_level_entity')
-    domain_entity = args.get('domain_entity')
-    concrete_entity = args.get('concrete_entity')
+            url = latest_schema["_links"]["json-schema"]["href"]
 
-    # flags
-    json_schema = 'json' in args
-    deref = 'deref' in args
-    latest = 'latest' in args
+        if json_schema and url and deref:
+            data = schema_service.get_dereferenced_schema(url)
+            return response_json(HTTPStatus.OK, data)
 
-    schema_service = SchemaService()
+        if json_schema and url and not deref:
+            data = schema_service.get_json_schema(url)
+            return response_json(HTTPStatus.OK, data)
 
-    if not url and latest:
-        result = app.ingest_api.get_schemas(
-            latest_only=latest,
-            high_level_entity=high_level_entity,
-            domain_entity=domain_entity,
-            concrete_entity=concrete_entity
-        )
+        return response_json(HTTPStatus.NOT_FOUND, None)
 
-        latest_schema = result[0] if len(result) > 0 else None
 
-        if not json_schema:
-            return response_json(HTTPStatus.OK, latest_schema)
+def create_app():
+    app = Flask(__name__, static_folder='static')
+    app.SPREADSHEET_STORAGE_DIR = os.environ.get('SPREADSHEET_STORAGE_DIR')
+    app.SPREADSHEET_UPLOAD_MESSAGE = "We’ve got your spreadsheet, and we’re currently importing and validating the data. \
+Nothing else for you to do - check back later."
 
-        url = latest_schema["_links"]["json-schema"]["href"]
+    app.SPREADSHEET_UPLOAD_MESSAGE_ERROR = "We experienced a problem while uploading your spreadsheet"
+    app.secret_key = 'cells'
+    cors = CORS(app, expose_headers=["Content-Disposition"])
+    app.config['CORS_HEADERS'] = 'Content-Type'
 
-    if json_schema and url and deref:
-        data = schema_service.get_dereferenced_schema(url)
-        return response_json(HTTPStatus.OK, data)
+    app.ingest_api = IngestApi()
+    spreadsheet_generator = SpreadsheetGenerator(app.ingest_api)
+    app.spreadsheet_job_manager = SpreadsheetJobManager(spreadsheet_generator, app.SPREADSHEET_STORAGE_DIR)
 
-    if json_schema and url and not deref:
-        data = schema_service.get_json_schema(url)
-        return response_json(HTTPStatus.OK, data)
+    app.register_blueprint(upload_bp)
+    app.register_blueprint(submissions_bp)
+    app.register_blueprint(import_geo_bp)
 
-    return response_json(HTTPStatus.NOT_FOUND, None)
+    define_app(app)
+    return app
 
 
 if __name__ == '__main__':
+    app = create_app()
```

Comment:
this looks like useful refactoring but one that is hard to review since comparing to the previous version is such a pain with things haven't moved around quite a bit :(

Reply:
You could hide the whitespace by adding w=1 in request param https://github.com/ebi-ait/ingest-broker/pull/26/files?w=1 or thru the PR review settings in the gear icon in the UI.